bindings_generator/src/api.rs

```diff
@@ -119,6 +125,67 @@ impl core::cmp::PartialOrd for Enum {
     }
 }
 
+impl Enum {
+    pub fn strip_common_prefix(&mut self) {
+        let mut variants: Vec<_> = self.values.iter().map(|v| (&v.0[..], *v.1)).collect();
+
+        // Build a map of prefix to occurrence
+        loop {
+            let underscore_index = variants[0].0.chars().enumerate().find_map(|(index, ch)| {
+                if ch == '_' {
+                    Some(index)
+                } else {
+                    None
+                }
+            });
+
+            let underscore_index = match underscore_index {
+                Some(index) if index > 0 => index,
+                Some(_) | None => break,
+            };
+
+            // Get a slice up to and including the `_`
+            let prefix = &variants[0].0[..=underscore_index];
+
+            let common_prefix = if variants.iter().all(|v| v.0.starts_with(prefix)) {
+                prefix
+            } else {
+                break;
+            };
```

Comment:
It might be clearer to write:

```rust
if !variants.iter().all(|v| v.0.starts_with(prefix)) {
    break;
}
```

...?

Reply:
That is quite a bit clearer!